### PR TITLE
Issue #503 - Link Stan function reference to pkgdown

### LINKED ISSUE #503: Prettify and link rendered `stan` docs to website
When #500 is merged, the next steps are to:

- [ ] Create a custom [index page](https://doxygen.nl/manual/additional.html)
- [x] Link the rendered docs to the package website
- [ ] Add a custom theme. See [comment](https://github.com/epinowcast/epinowcast/issues/503#issuecomment-2298649695).

### PR DESCRIPTION
**Pull request by jamesmbaazam**
## Description

This PR partially closes #503.

[Describe the changes that you made in this pull request.]

This PR restructures the pkgdown yml configuration and links the Stan documentation.

## Checklist

- [x] My PR is based on a package issue and I have explicitly linked it.
- [x] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [x] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [x] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [x] My code follows the established coding standards.
- [x] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 ## Documentation
 
+- Linked the Stan function documentation to the package website. By @jamesmbaazam in # and reviewed by @seabbs.
```
Comment (seabbs) [inline NEWS.md:35]:
```suggestion
- Linked the Stan function documentation to the package website. By @jamesmbaazam in #529 and reviewed by @seabbs.
```

**Review by seabbs — DISMISSED** [review]
This is really nice and I'm happy to mergeee - thanks @jamesmbaazam! However, I have unlinked the issue as I don't think this closes it (we still need a custom homepage for the stan code and a prettier theme).

**Comment by jamesmbaazam** [comment]
Ooopsie! I was going to fix the other bullet points in the issue in this PR but it's fine that you've merged it. It has a nice small scope. I'll tackle the others in a separate PR.

**Comment by seabbs** [comment]
Yes no I agree I liked the tight scope of this

**Pull request merged**